### ISSUE TITLE
mape loss if the lable is 0

### ISSUE THREAD

**Issue by tangtangsiqi** [fchollet/keras#2908]
Please make sure that the boxes below are checked before you submit your issue. Thank you!
- [ ] Check that you are up-to-date with the master branch of Keras. You can update with:
  pip install git+git://github.com/fchollet/keras.git --upgrade --no-deps
- [ ] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
  pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps
- [ ] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

**Comment by tangtangsiqi** [comment]
in my project the mean absolute percentage error is used, but when the lable is 0, I think the error is not  important and i dont count it. so i try to write another mape loss function based on the existing mape. however it doesnt work ,i dont know how to form an index and how to set the value of a tensorvariable using an index.


def error_tsq_mape(y_true, y_pred):
    diff = K.abs((y_true - y_pred) / K.clip(K.abs(y_true), K.epsilon(), np.inf))
    ind=K.zeros_like(y_true)
    print 'a'
    if K.not_equal(K.ndim(ind),0):
        diff[ind]=0
    return 100. * K.mean(diff, axis=-1)
